src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java

```diff
@@ -0,0 +1,150 @@
+package org.rulez.demokracia.pdengine;
+
+import static org.junit.Assert.assertEquals;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.Test;
+import org.rulez.demokracia.pdengine.annotations.tested_behaviour;
+import org.rulez.demokracia.pdengine.annotations.tested_feature;
+import org.rulez.demokracia.pdengine.annotations.tested_operation;
+import org.rulez.demokracia.pdengine.exception.ReportedException;
+import org.rulez.demokracia.pdengine.testhelpers.CreatedDefaultVoteRegistry;
+
+public class VoteShowTest extends CreatedDefaultVoteRegistry {
+
+	Vote vote;
+	String voteId;
+	String adminKey;
+	
+	public void setUp() throws ReportedException {
+		super.setUp();
+		vote = voteManager.getVote(adminInfo.voteId);
+		vote.addChoice("choiceName", "user");
+		voteManager.addChoice(adminInfo.adminKey, adminInfo.voteId, "choiceName", "user");
+		voteId = adminInfo.voteId;
+		adminKey = adminInfo.adminKey;
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void the_name_attribute_contains_the_name_of_the_vote() throws ReportedException {
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("name"), vote.name);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void the_canAddIn_attribute_contains_the_voters_can_add_choices_to_the_vote() throws ReportedException {
```

Comment:
at least a "whether" is missing here.
tha canAddin attribute contains whether the voters can add choices to the vote

Reply:
Fixed

Reply:
No problem. I would like to be free. :) 
And I would like to give you high quality code and documentation.